### DISCUSSION TITLE
.NET analyzers complain about the Dispose method in Blazor Component

### DISCUSSION THREAD

**Discussion by TanvirArjel** [dotnet/aspnetcore#28952] [Q&A]
ASP.NET Core Blazor team suggested `Dispose` method in a Blazor Component is as follows:

```
public partial class Products : IDisposable
{
      public void Dispose()
      {
          _navigationManager.LocationChanged -= NavigationManager_LocationChanged;
      }
}
```

The above code throws the following warnings:

>warning CA1816: Change Products.Dispose() to call GC.SuppressFinalize(object). This will prevent derived types that introduce a finalizer from needing to re-implement 'IDisposable' to call it.

> warning CA1063: Modify 'Products.Dispose' so that it calls Dispose(true), then calls GC.SuppressFinalize on the current object instance ('this' or 'Me' in Visual Basic), and then returns

If I write the `Dispose` method as follows then the warnings go away:

    public void Dispose()
    {
          Dispose(true);
          GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool disposing)
    {
        if (disposing)
        {
             _navigationManager.LocationChanged -= NavigationManager_LocationChanged;
        }
    }

My question is: **Is it worthwhile calling `GC.SuppressFinalize(this);` in Blazor Component?**

**Comment by renanrcp** [comment]
For the .NET vision, every Blazor component inherits from ComponentBase, and ComponentBase doesn't have a destructor, with this information I think you can safely call `GC.SuppressFinalize(this);`
